src/gui/virtualstudio.cpp

```diff
@@ -1837,41 +832,28 @@ void VirtualStudio::disconnect()
     m_connectionState = QStringLiteral("Disconnected");
     emit connectionStateChanged();
 
-    // cleanup
-    m_monitor            = nullptr;
-    m_inputMeter         = nullptr;
-    m_outputMeter        = nullptr;
-    m_inputVolumePlugin  = nullptr;
-    m_outputVolumePlugin = nullptr;
-    m_currentStudio      = -1;
-
-    if (currentStudioID != QStringLiteral("")) {
-        openFeedbackModal(currentStudioID);
+    if (!m_currentStudio.id().isEmpty()) {
+        emit openFeedbackSurveyModal(m_currentStudio.id());
     }
+
+    // cleanup
+    m_currentStudio.setId("");
 }
 
-void VirtualStudio::manageStudio(int studioIndex, bool start)
+void VirtualStudio::manageStudio(const QString& studioId, bool start)
```

Comment:
I have a feeling this function is pretty much deprecated with the 2.0 work, but in any case it might be helpful to have some guard clause in case `studioId.isEmpty()`?

Reply:
Added checks